com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/ResolveClasspathsHandler.java

```diff
@@ -302,7 +328,18 @@ protected JavaApplicationLaunchConfiguration(IProject project, String mainType,
             } else if (ProjectUtils.isGradleProject(project)) {
                 classpathProvider = "org.eclipse.buildship.core.classpathprovider";
             }
-            this.launchInfo = new JavaLaunchConfigurationInfo(JAVA_APPLICATION_LAUNCH);
+
+            // Since MavenRuntimeClasspathProvider will only including test entries when:
+            // 1. Launch configuration is JUnit/TestNG type
+            // 2. Mapped resource is in test path.
+            // That's why we use JUnit launch configuration here to make sure the result is right when excludeTestCode is false.
+            String launchXml = null;
+            if (!excludeTestCode && mappedResources.isEmpty()) {
+                launchXml = String.format(JAVA_APPLICATION_LAUNCH, "org.eclipse.jdt.junit.launchconfig");
```

Comment:
if test runner extension is not installed, does this still work?

Reply:
Yes it is. junit is already embedded in JLS.